### PR TITLE
add CRUD for account

### PR DESCRIPTION
**Pull request by msm-faraj**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
 const express = require("express");
-const process = require("process");
+// const process = require("process"); //???
```
Comment (pouyapourbaba) [inline src/index.js:2]: What is this?

File: src/schema.js
Comment (pouyapourbaba) [inline src/schema.js:1]: Separate the schema for each controller into its own file

**Review by pouyapourbaba — APPROVED** [review]

File: src/controllers/account/accountController.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      name,
+      description,
+      userId: req.authenticatedUser.id,
+      cash,
+      currency,
+    });
+    return res.status(httpStatus.CREATED).send(account);
+  }
+
+  async findAllAccounts(req, res) {
```
Comment (pouyapourbaba) [inline src/controllers/account/accountController.js:30]: We need to return only the accounts for user who sends the request

File: src/controllers/account/accountController.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    if (!account) {
+      return res.status(httpStatus.NOT_FOUND).send("Account not found");
+    }
+    return res.status(httpStatus.OK).send(account);
+  }
+
+  async updateAccount(req, res) {
+    const { name, description, cash, currency } = req.body;
+
+    const [rowsUpdated, [updatedAccount]] = await this.Account.update(
```
Comment (pouyapourbaba) [inline src/controllers/account/accountController.js:48]: What will happen if the name and description are not in the request body?

File: src/controllers/account/accountController.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        description,
+        cash,
+        currency,
+      },
+      {
+        where: { id: req.params.id },
+        returning: true,
+      }
+    );
+    if (rowsUpdated === 0) {
```
Comment (pouyapourbaba) [inline src/controllers/account/accountController.js:60]: Let's find the account first and do the 404 check before updating the record.

File: src/controllers/account/accountController.js (hunk 2 above)
Comment (msm-faraj) [reply src/controllers/account/accountController.js:30]: done

File: src/controllers/account/accountController.js (hunk 4 above)
Comment (msm-faraj) [reply src/controllers/account/accountController.js:60]: done

File: src/controllers/account/accountController.js (hunk 3 above)
Comment (msm-faraj) [reply src/controllers/account/accountController.js:48]: It works as we expected
It only updates the properties whic are in the req.body and don't change the others

File: src/schema.js
Comment (msm-faraj) [reply src/schema.js:1]: done

**Review by pouyapourbaba — APPROVED** [review]

**Pull request merged**